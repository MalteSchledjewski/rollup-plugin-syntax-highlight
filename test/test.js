import { test } from "uvu";
import { resolve } from "path";

import { rollup } from "rollup";
import { syntaxHighlight } from "../dist/index.mjs";
import { fileSnapshot } from "./utils.js";
import { writeFile } from "fs/promises";

const outputs = [];

async function testFixture(fixture) {
  const bundle = await rollup({
    input: resolve(`test/fixtures/${fixture}?syntax`),
    plugins: [syntaxHighlight()],
  });

  const { output } = await bundle.generate({
    file: "file.js",
  });

  const code = output[0].code;

  await fileSnapshot(fixture, code);

  outputs.push({ name: fixture, code });
}

test("css", async () => {
  await testFixture("test.css");
});

test("html", async () => {
  await testFixture("test.html");
});

test("js", async () => {
  await testFixture("test.js");
});

test("rs", async () => {
  await testFixture("test.rs");
});

test("svelte", async () => {
  await testFixture("test.svelte");
});

test("svg", async () => {
  await testFixture("test.svg");
});

test("ts", async () => {
  await testFixture("test.ts");
});

test.after(async () => {
  const texturesHtml = outputs
    .map(({ name, code }) => {
      const start = code.indexOf('"<pre');
      const end = code.lastIndexOf('/pre>"') + 6;
      const html = JSON.parse(code.slice(start, end));

      return `<h2>${name}</h2>\n` + html;
    })
    .join("\n\n");

  // for creating demo.html

  const html = `
<html>
    <body>
${texturesHtml}
    </body>
</html>`;

  await writeFile("test/demo.html", html, {
    encoding: "utf-8",
  });
});

test.run();
