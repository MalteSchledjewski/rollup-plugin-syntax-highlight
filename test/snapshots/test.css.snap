var test_css = "<pre class=\"shiki\" style=\"background-color: #1E1E1E\"><code><span class=\"line\"><span style=\"color: #D7BA7D\">.test</span><span style=\"color: #D4D4D4\"> {</span></span>\n<span class=\"line\"><span style=\"color: #D4D4D4\">  </span><span style=\"color: #9CDCFE\">background-color</span><span style=\"color: #D4D4D4\">: </span><span style=\"color: #CE9178\">red</span><span style=\"color: #D4D4D4\">;</span></span>\n<span class=\"line\"><span style=\"color: #D4D4D4\">}</span></span></code></pre>";

export { test_css as default };
